src/loader/loader.js

```diff
@@ -49,6 +49,20 @@ export default class Loader {
 		const config = this._config;
 
 		const name = args[0];
+
+		let module = config.getModule(name);
+
+		if (module && module.defined) {
+			if (config.showWarnings) {
+				console.warn(
+					`Liferay AMD Loader: Module '${name}' is being ` +
+						'redefined; only the first definition will be used'
+				);
```

Comment:
Would it be overkill to only warn once per `name`? (ie. stick the warned names in a `new Set()` and warn only `if (config.showWarnings && !shownWarnings.has(name))`)

Or were you thinking it will be useful to see the stack traces for multiple duplicates?

Or perhaps that this will be so rare as to not have to worry about it?

Reply:
Well, if you want to get notified of redefinitions you probably want to know all of them so, if we dedupe, the code will be more complex and probably less useful.

Also keep in mind that this won't show unless the user explicitly requests it (it is off by default) so there's not too much value in making it smarter, I think.